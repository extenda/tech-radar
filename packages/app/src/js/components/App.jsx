import { jwtDecode } from 'jwt-decode';
import { withLDConsumer } from 'launchdarkly-react-client-sdk';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import radarService from '../modules/radarService';
import sha256 from '../modules/sha256';
import Entry from './Entry';
import Footer from './Footer';
import Login from './Login';
import Logout from './Logout';
import NotFound from './NotFound';
import Quadrant from './Quadrant';
import LDRadar from './Radar';
import TagList from './TagList';

export class App extends Component {
  constructor(props) {
    super(props);
    if (typeof DISABLE_GOOGLE_LOGIN === 'boolean' && DISABLE_GOOGLE_LOGIN) {
      // No log in required. Usually desired in development server.
      this.state = {
        isSignedIn: true,
        loading: true,
      };
      radarService.init().then(this.radarDidLoad);
    } else {
      this.state = {
        isSignedIn: false,
        loading: true,
      };
    }
  }

  radarDidLoad = () => {
    this.setState((prevState) => ({
      ...prevState,
      loading: false,
    }));
  };

  radarDidFail = (err) => {
    // eslint-disable-next-line no-console
    console.error('Failed to load radar entries', err);
    this.setState((prevState) => ({
      ...prevState,
      isSignedIn: false,
      loading: true,
    }));
  };

  loginDidSucceed = (response) => {
    this.setState((prevState) => ({
      ...prevState,
      isSignedIn: true,
    }));

    const { ldClient } = this.props;
    if (!ldClient) {
      // eslint-disable-next-line no-console
      console.debug('Missing LDClient property');
      return null;
    }

    const { sub, email } = jwtDecode(response.credential);
    return sha256(sub).then((key) =>
      ldClient
        .identify({
          kind: 'user',
          key,
          email,
          _meta: {
            privateAttributeNames: ['email'],
          },
        })
        .then(() =>
          radarService.init(response.credential).then(this.radarDidLoad).catch(this.radarDidFail),
        ),
    );
  };

  loginDidFail = () => {};

  render = () => {
    const { loading, isSignedIn } = this.state;

    if (!isSignedIn) {
      return <Login onSuccess={this.loginDidSucceed} onFailure={this.loginDidFail} />;
    }

    if (loading) {
      return null;
    }

    return (
      <>
        <Logout />
        <BrowserRouter>
          <Switch>
            <Route exact path="/" component={LDRadar} />
            <Route path="/entries/:id" component={Entry} />
            <Route path="/:quadrant.html" component={Quadrant} />
            <Route path="/tags/:tag.html" component={TagList} />
            <Route path="*" component={NotFound} />
          </Switch>
        </BrowserRouter>
        <Footer />
      </>
    );
  };
}

App.propTypes = {
  ldClient: PropTypes.shape({
    identify: PropTypes.func.isRequired,
  }),
};
App.defaultProps = {
  ldClient: null,
};

export default withLDConsumer({ clientOnly: true })(App);
