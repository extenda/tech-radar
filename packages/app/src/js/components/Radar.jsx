import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withLDConsumer } from 'launchdarkly-react-client-sdk';
import { pick } from '@tech-radar/shared/src/utils';
import SvgRadar from '../lib/radar';
import radarService from '../modules/radarService';
import Icon from './Icon';
import Navigation from './Navigation';
import TagsInput from './TagsInput';
import QuadrantList from './QuadrantList';

export class Radar extends Component {
  constructor(props) {
    super(props);
    this.svgRadar = this.createSvgRadar();
    this.state = {
      blips: radarService.listBlips([]),
      tags: radarService.listTags(),
    };
  }

  createSvgRadar = () => {
    const { history } = this.props;

    const radar = radarService.model;

    return new SvgRadar({
      svg_id: 'radar',
      width: 900, // 1450 Set to 900 if not printing labels.
      height: 900,
      colors: {
        background: '#fff',
        grid: '#bbb',
        inactive: '#ddd',
      },
      quadrants: radar.quadrants.map((q) => pick(q, 'name')),
      rings: [
        { name: 'Adopt', color: '#43a047' },
        { name: 'Trial', color: '#c0ca33' },
        { name: 'Assess', color: '#ffb300' },
        { name: 'Hold', color: '#bf360c' },
      ],
      print_layout: true,
      print_entities: false,
      print_footer: true,
      linkOnClick: (link) => {
        history.push(link);
      },
    });
  };

  renderRadar = () => {
    this.svgRadar.renderBackground();
    this.renderBlips();
  };

  renderBlips = () => {
    const { blips } = this.state;
    this.svgRadar.renderEntries(blips);
  };

  onFilter = (tags) => {
    this.setState((prevState) => ({
      ...prevState,
      blips: radarService.listBlips(tags.map((t) => t.name)),
    }), this.renderBlips);
  };

  onChangeRadar = (e) => {
    // We must trigger a change on the TagFilter component to update the blips!
    const id = e.currentTarget.getAttribute('data-radar-id');
    if (radarService.model.id !== id) {
      radarService.switchRadar(id).then(() => {
        this.setState((prevState) => ({
          ...prevState,
          tags: radarService.listTags(),
          blips: radarService.listBlips(),
        }), this.renderBlips);
      });
    }
  }

  componentDidMount = () => {
    this.renderRadar();
  };

  onMouseEnter = (e) => {
    const id = e.currentTarget.getAttribute('data-radar-id');
    this.svgRadar.showBubbleForId(id);
    this.svgRadar.highlightLegendItem({ id });
  };

  onMouseLeave = (e) => {
    const id = e.currentTarget.getAttribute('data-radar-id');
    this.svgRadar.hideBubble();
    this.svgRadar.unhighlightLegendItem({ id });
  };

  renderQuadrant = (quadrant) => {
    const { blips } = this.state;
    const entries = radarService.getQuadrant(quadrant.dirname, true);

    return (
      <>
        <h2>{quadrant.name}</h2>
        <QuadrantList
          quadrant={entries}
          blips={blips}
          headerLevel={3}
          useShortname
          onMouseEnter={this.onMouseEnter}
          onMouseLeave={this.onMouseLeave}
        />
      </>
    );
  }

  render = () => {
    const radar = radarService.model;

    const { flags } = this.props;
    const { tags } = this.state;

    return (
      <>
        <Navigation home={radar.quadrantsNavBar} />
        {flags.releaseToolRadar && (
          <>
            <div className="u-full-width" style={{ marginBottom: '20px' }}>
              <center>
                <button type="button" data-radar-id="radar" onClick={this.onChangeRadar}>
                  <Icon name="rss" />
                  Engineering
                </button>
                <button type="button" data-radar-id="radar_it" onClick={this.onChangeRadar}>
                  <Icon name="signal" />
                  IT & BIS
                </button>
              </center>
            </div>
            <h1 className="center">
              {radar.title}
              &nbsp;
              <span className="version">
                <Icon name="calendar" />
                {radar.formattedDate}
              </span>
            </h1>
          </>
        )}
        {!flags.releaseToolRadar && (
        <h1 className="center">
          {radar.title}
          &nbsp;
          <span className="version">
            <Icon name="tag" />
            {radar.version}
          </span>
        </h1>
        )}
        <div className="tags-filter">
          <TagsInput onFilter={this.onFilter} tags={tags} />
        </div>
        <div className="radar-container u-full-width">
          <div className="radar-entries">
            {this.renderQuadrant(radar.quadrants[2])}
            {this.renderQuadrant(radar.quadrants[1])}
          </div>
          <div className="svg-radar">
            <svg id="radar" />
          </div>
          <div className="radar-entries">
            {this.renderQuadrant(radar.quadrants[3])}
            {this.renderQuadrant(radar.quadrants[0])}
          </div>
        </div>
        <div className="u-cf" />
        <div className="container">
          <div className="row">
            <div className="one-half column">
              <h2>About the Tech Radar</h2>
              <p>
                The Extenda Retail Tech Radar is based on the concept introduced by
                {' '}
                <a href="https://www.thoughtworks.com/radar">ThoughtWorks</a>
                .
                The Radar contains a list of technologies, complemented by an assessment result,
                called ring assessment.
                The following rings are used:
              </p>
              <ul className="no-bullets">
                <li>
                  <strong>Adopt</strong>
                  {' — '}
                  Technologies we have high confidence in and recommend for wide use.
                </li>
                <li>
                  <strong>Trial</strong>
                  {' — '}
                  Technologies we have seen used with success, but are yet to be proved successfully
                  in wide use and at large scale. Carries higher risk.
                </li>
                <li>
                  <strong>Assess</strong>
                  {' — '}
                  Technologies that are promising with a clear potential value for us.
                  Requires research and prototyping to analyze its impact.
                </li>
                <li>
                  <strong>Hold</strong>
                  {' — '}
                  Technologies not recommended for new products or technology we no longer
                  want to invest in. Existing products may still consider these depending
                  on their legacy.
                </li>
              </ul>
            </div>
            <div className="one-half column">
              <h2>The Purpose</h2>
              <p>
                The Tech Radar aims to inspire and support teams at Extenda Retail to pick the
                best technologies for their projects. It is way to share experience and knowledge
                between the teams and to create transparency about the technology direction of
                Extenda Retail.
                The Tech Radar also becomes a list of dos and don&apos;ts, what to try and to avoid
                in order to increase success.
              </p>
              <h2>How to Contribute</h2>
              <p>
                The Tech Radar is maintained on
                {' '}
                <a href="https://github.com/extenda/tech-radar">GitHub</a>
                .
                There you can read more about the process and how to contribute.
              </p>
            </div>
          </div>
        </div>
      </>
    );
  };
}

Radar.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  flags: PropTypes.object,
};

Radar.defaultProps = {
  flags: {},
};

export default withLDConsumer()(Radar);
