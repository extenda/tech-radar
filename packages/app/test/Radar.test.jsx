import { mount, shallow } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import { Radar } from '../src/js/components/Radar';

jest.mock('../src/js/modules/radarService');

describe('<Radar />', () => {
  beforeAll(() => {
    const div = document.createElement('div');
    window.domNode = div;
    document.body.appendChild(div);
  });

  afterAll(() => {
    document.body.removeChild(window.domNode);
    window.domNode = null;
  });

  test('It renders the SVG radar and description', () => {
    const component = shallow(<Radar history={{ push: jest.fn() }} />);
    expect(component).toMatchSnapshot();
  });

  test('It renders the radar blips in SVG', () => {
    const component = mount(
      <MemoryRouter>
        <Radar history={{ push: jest.fn() }} />
      </MemoryRouter>,
      { attachTo: window.domNode },
    );
    expect(document.querySelectorAll('g.blip')).toHaveLength(2);
    component.unmount();
  });

  test('It renders IT & BIS radar switch', () => {
    const component = shallow(
      <Radar
        history={{ push: jest.fn() }}
        flags={{ releaseToolRadar: true }}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  test('It is possible to click radar blips', () => {
    const clickFn = jest.fn();
    const component = mount(
      <MemoryRouter>
        <Radar history={{ push: clickFn }} />
      </MemoryRouter>,
      { attachTo: window.domNode },
    );
    document.querySelectorAll('g.blip')[0].dispatchEvent(new Event('click'));
    expect(clickFn.mock.calls.length).toEqual(1);
    component.unmount();
  });

  test('It reacts to mouse events on quadrants listings', () => {
    const component = mount(
      <MemoryRouter>
        <Radar history={{ push: jest.fn() }} />
      </MemoryRouter>,
      { attachTo: window.domNode },
    );

    const showBubble = jest
      .spyOn(component.find(Radar).instance().svgRadar, 'showBubble')
      .mockImplementationOnce(() => {});

    const legendItem = component.find('#legendItem1').at(0);

    // Simulate a mouse enter from the DOM element
    legendItem.find('a').at(0).simulate('mouseEnter');

    expect(showBubble).toHaveBeenCalled();
    expect(legendItem.getDOMNode().className).toContain('radar-highlight');

    // Simulate a mouse leave from the DOM element
    legendItem.find('a').at(0).simulate('mouseLeave');

    expect(legendItem.getDOMNode().className).not.toContain('radar-highlight');

    component.unmount();
    showBubble.mockRestore();
  });

  test('It is possible to filter the radar blips', () => {
    const component = mount(
      <MemoryRouter>
        <Radar history={{ push: jest.fn() }} />
      </MemoryRouter>,
      { attachTo: window.domNode },
    );

    expect(document.querySelectorAll('g.blip')).toHaveLength(2);

    // Apply filter
    const radar = component.find(Radar);
    radar.instance().onFilter([{ id: 1, name: 'java' }]);

    expect(document.querySelectorAll('g.blip')).toHaveLength(1);

    component.unmount();
  });
});
