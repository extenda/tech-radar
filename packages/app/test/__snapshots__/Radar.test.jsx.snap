// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radar /> It renders IT & BIS radar switch 1`] = `
<Fragment>
  <Navigation
    entry={null}
    home={
      [
        {
          "dirname": "test1",
          "name": "Test 1",
        },
        {
          "dirname": "test2",
          "name": "Test 2",
        },
        {
          "dirname": "dev",
          "name": "Development",
        },
        {
          "dirname": "test4",
          "name": "Test 4",
        },
      ]
    }
    quadrant={null}
    tag={null}
  />
  <div
    className="u-full-width"
    style={
      {
        "marginBottom": "20px",
      }
    }
  >
    <center>
      <button
        data-radar-id="radar"
        onClick={[Function]}
        type="button"
      >
        <Icon
          name="rss"
          spaceBefore={true}
        />
        Engineering
      </button>
      <button
        data-radar-id="radar_it"
        onClick={[Function]}
        type="button"
      >
        <Icon
          name="signal"
          spaceBefore={true}
        />
        IT & BIS
      </button>
    </center>
  </div>
  <h1
    className="center"
  >
    Test
    Â 
    <span
      className="version"
    >
      <Icon
        name="calendar"
        spaceBefore={true}
      />
      5 June, 2019
    </span>
  </h1>
  <div
    className="tags-filter"
  >
    <TagsInput
      onFilter={[Function]}
      tags={
        [
          "90's",
          "java",
        ]
      }
    />
  </div>
  <div
    className="radar-container u-full-width"
  >
    <div
      className="radar-entries"
    >
      <h2>
        Development
      </h2>
      <QuadrantList
        blips={
          [
            {
              "active": true,
              "color": "#43a047",
              "id": 1,
              "index": 0,
              "label": "Java",
              "link": "/entries/java.html",
              "moved": false,
              "quadrant": 2,
              "ring": 0,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -14.474380723221014,
              "y": -41.11798108601353,
            },
            {
              "active": true,
              "color": "#bf360c",
              "id": 3,
              "index": 1,
              "label": "PHP",
              "link": "/entries/php.html",
              "moved": false,
              "quadrant": 2,
              "ring": 3,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -382.075380904692,
              "y": -21.17457418758074,
            },
          ]
        }
        headerLevel={3}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        quadrant={
          {
            "adopt": [
              {
                "active": true,
                "filename": "1.html",
                "id": 1,
                "name": "A - 1",
              },
              {
                "active": true,
                "filename": "2.html",
                "id": 2,
                "name": "B - 2",
              },
            ],
            "assess": [
              {
                "active": true,
                "filename": "3.html",
                "id": 3,
                "name": "C - 3",
              },
            ],
            "dirname": "test1",
            "hold": [
              {
                "active": true,
                "filename": "4.html",
                "id": 4,
                "name": "D - 4",
              },
              {
                "active": true,
                "filename": "5.html",
                "id": 5,
                "name": "E - 5",
              },
              {
                "active": false,
                "filename": "6.html",
                "id": 6,
                "name": "F - 6",
              },
            ],
            "name": "Test",
            "trial": [],
          }
        }
        useShortname={true}
      />
      <h2>
        Test 2
      </h2>
      <QuadrantList
        blips={
          [
            {
              "active": true,
              "color": "#43a047",
              "id": 1,
              "index": 0,
              "label": "Java",
              "link": "/entries/java.html",
              "moved": false,
              "quadrant": 2,
              "ring": 0,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -14.474380723221014,
              "y": -41.11798108601353,
            },
            {
              "active": true,
              "color": "#bf360c",
              "id": 3,
              "index": 1,
              "label": "PHP",
              "link": "/entries/php.html",
              "moved": false,
              "quadrant": 2,
              "ring": 3,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -382.075380904692,
              "y": -21.17457418758074,
            },
          ]
        }
        headerLevel={3}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        quadrant={
          {
            "adopt": [
              {
                "active": true,
                "filename": "1.html",
                "id": 1,
                "name": "A - 1",
              },
              {
                "active": true,
                "filename": "2.html",
                "id": 2,
                "name": "B - 2",
              },
            ],
            "assess": [
              {
                "active": true,
                "filename": "3.html",
                "id": 3,
                "name": "C - 3",
              },
            ],
            "dirname": "test1",
            "hold": [
              {
                "active": true,
                "filename": "4.html",
                "id": 4,
                "name": "D - 4",
              },
              {
                "active": true,
                "filename": "5.html",
                "id": 5,
                "name": "E - 5",
              },
              {
                "active": false,
                "filename": "6.html",
                "id": 6,
                "name": "F - 6",
              },
            ],
            "name": "Test",
            "trial": [],
          }
        }
        useShortname={true}
      />
    </div>
    <div
      className="svg-radar"
    >
      <svg
        id="radar"
      />
    </div>
    <div
      className="radar-entries"
    >
      <h2>
        Test 4
      </h2>
      <QuadrantList
        blips={
          [
            {
              "active": true,
              "color": "#43a047",
              "id": 1,
              "index": 0,
              "label": "Java",
              "link": "/entries/java.html",
              "moved": false,
              "quadrant": 2,
              "ring": 0,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -14.474380723221014,
              "y": -41.11798108601353,
            },
            {
              "active": true,
              "color": "#bf360c",
              "id": 3,
              "index": 1,
              "label": "PHP",
              "link": "/entries/php.html",
              "moved": false,
              "quadrant": 2,
              "ring": 3,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -382.075380904692,
              "y": -21.17457418758074,
            },
          ]
        }
        headerLevel={3}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        quadrant={
          {
            "adopt": [
              {
                "active": true,
                "filename": "1.html",
                "id": 1,
                "name": "A - 1",
              },
              {
                "active": true,
                "filename": "2.html",
                "id": 2,
                "name": "B - 2",
              },
            ],
            "assess": [
              {
                "active": true,
                "filename": "3.html",
                "id": 3,
                "name": "C - 3",
              },
            ],
            "dirname": "test1",
            "hold": [
              {
                "active": true,
                "filename": "4.html",
                "id": 4,
                "name": "D - 4",
              },
              {
                "active": true,
                "filename": "5.html",
                "id": 5,
                "name": "E - 5",
              },
              {
                "active": false,
                "filename": "6.html",
                "id": 6,
                "name": "F - 6",
              },
            ],
            "name": "Test",
            "trial": [],
          }
        }
        useShortname={true}
      />
      <h2>
        Test 1
      </h2>
      <QuadrantList
        blips={
          [
            {
              "active": true,
              "color": "#43a047",
              "id": 1,
              "index": 0,
              "label": "Java",
              "link": "/entries/java.html",
              "moved": false,
              "quadrant": 2,
              "ring": 0,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -14.474380723221014,
              "y": -41.11798108601353,
            },
            {
              "active": true,
              "color": "#bf360c",
              "id": 3,
              "index": 1,
              "label": "PHP",
              "link": "/entries/php.html",
              "moved": false,
              "quadrant": 2,
              "ring": 3,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -382.075380904692,
              "y": -21.17457418758074,
            },
          ]
        }
        headerLevel={3}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        quadrant={
          {
            "adopt": [
              {
                "active": true,
                "filename": "1.html",
                "id": 1,
                "name": "A - 1",
              },
              {
                "active": true,
                "filename": "2.html",
                "id": 2,
                "name": "B - 2",
              },
            ],
            "assess": [
              {
                "active": true,
                "filename": "3.html",
                "id": 3,
                "name": "C - 3",
              },
            ],
            "dirname": "test1",
            "hold": [
              {
                "active": true,
                "filename": "4.html",
                "id": 4,
                "name": "D - 4",
              },
              {
                "active": true,
                "filename": "5.html",
                "id": 5,
                "name": "E - 5",
              },
              {
                "active": false,
                "filename": "6.html",
                "id": 6,
                "name": "F - 6",
              },
            ],
            "name": "Test",
            "trial": [],
          }
        }
        useShortname={true}
      />
    </div>
  </div>
  <div
    className="u-cf"
  />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="one-half column"
      >
        <h2>
          About the Tech Radar
        </h2>
        <p>
          The Extenda Retail Tech Radar is based on the concept introduced by 
          <a
            href="https://www.thoughtworks.com/radar"
          >
            ThoughtWorks
          </a>
          . The Radar contains a list of technologies, complemented by an assessment result, called ring assessment. The following rings are used:
        </p>
        <ul
          className="no-bullets"
        >
          <li>
            <strong>
              Adopt
            </strong>
             â 
            Technologies we have high confidence in and recommend for wide use.
          </li>
          <li>
            <strong>
              Trial
            </strong>
             â 
            Technologies we have seen used with success, but are yet to be proved successfully in wide use and at large scale. Carries higher risk.
          </li>
          <li>
            <strong>
              Assess
            </strong>
             â 
            Technologies that are promising with a clear potential value for us. Requires research and prototyping to analyze its impact.
          </li>
          <li>
            <strong>
              Hold
            </strong>
             â 
            Technologies not recommended for new products or technology we no longer want to invest in. Existing products may still consider these depending on their legacy.
          </li>
        </ul>
      </div>
      <div
        className="one-half column"
      >
        <h2>
          The Purpose
        </h2>
        <p>
          The Tech Radar aims to inspire and support teams at Extenda Retail to pick the best technologies for their projects. It is way to share experience and knowledge between the teams and to create transparency about the technology direction of Extenda Retail. The Tech Radar also becomes a list of dos and don'ts, what to try and to avoid in order to increase success.
        </p>
        <h2>
          How to Contribute
        </h2>
        <p>
          The Tech Radar is maintained on
           
          <a
            href="https://github.com/extenda/tech-radar"
          >
            GitHub
          </a>
          . There you can read more about the process and how to contribute.
        </p>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`<Radar /> It renders the SVG radar and description 1`] = `
<Fragment>
  <Navigation
    entry={null}
    home={
      [
        {
          "dirname": "test1",
          "name": "Test 1",
        },
        {
          "dirname": "test2",
          "name": "Test 2",
        },
        {
          "dirname": "dev",
          "name": "Development",
        },
        {
          "dirname": "test4",
          "name": "Test 4",
        },
      ]
    }
    quadrant={null}
    tag={null}
  />
  <h1
    className="center"
  >
    Test
    Â 
    <span
      className="version"
    >
      <Icon
        name="tag"
        spaceBefore={true}
      />
      0.0.1-local
    </span>
  </h1>
  <div
    className="tags-filter"
  >
    <TagsInput
      onFilter={[Function]}
      tags={
        [
          "90's",
          "java",
        ]
      }
    />
  </div>
  <div
    className="radar-container u-full-width"
  >
    <div
      className="radar-entries"
    >
      <h2>
        Development
      </h2>
      <QuadrantList
        blips={
          [
            {
              "active": true,
              "color": "#43a047",
              "id": 1,
              "index": 0,
              "label": "Java",
              "link": "/entries/java.html",
              "moved": false,
              "quadrant": 2,
              "ring": 0,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -14.474380723221014,
              "y": -41.11798108601353,
            },
            {
              "active": true,
              "color": "#bf360c",
              "id": 3,
              "index": 1,
              "label": "PHP",
              "link": "/entries/php.html",
              "moved": false,
              "quadrant": 2,
              "ring": 3,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -382.075380904692,
              "y": -21.17457418758074,
            },
          ]
        }
        headerLevel={3}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        quadrant={
          {
            "adopt": [
              {
                "active": true,
                "filename": "1.html",
                "id": 1,
                "name": "A - 1",
              },
              {
                "active": true,
                "filename": "2.html",
                "id": 2,
                "name": "B - 2",
              },
            ],
            "assess": [
              {
                "active": true,
                "filename": "3.html",
                "id": 3,
                "name": "C - 3",
              },
            ],
            "dirname": "test1",
            "hold": [
              {
                "active": true,
                "filename": "4.html",
                "id": 4,
                "name": "D - 4",
              },
              {
                "active": true,
                "filename": "5.html",
                "id": 5,
                "name": "E - 5",
              },
              {
                "active": false,
                "filename": "6.html",
                "id": 6,
                "name": "F - 6",
              },
            ],
            "name": "Test",
            "trial": [],
          }
        }
        useShortname={true}
      />
      <h2>
        Test 2
      </h2>
      <QuadrantList
        blips={
          [
            {
              "active": true,
              "color": "#43a047",
              "id": 1,
              "index": 0,
              "label": "Java",
              "link": "/entries/java.html",
              "moved": false,
              "quadrant": 2,
              "ring": 0,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -14.474380723221014,
              "y": -41.11798108601353,
            },
            {
              "active": true,
              "color": "#bf360c",
              "id": 3,
              "index": 1,
              "label": "PHP",
              "link": "/entries/php.html",
              "moved": false,
              "quadrant": 2,
              "ring": 3,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -382.075380904692,
              "y": -21.17457418758074,
            },
          ]
        }
        headerLevel={3}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        quadrant={
          {
            "adopt": [
              {
                "active": true,
                "filename": "1.html",
                "id": 1,
                "name": "A - 1",
              },
              {
                "active": true,
                "filename": "2.html",
                "id": 2,
                "name": "B - 2",
              },
            ],
            "assess": [
              {
                "active": true,
                "filename": "3.html",
                "id": 3,
                "name": "C - 3",
              },
            ],
            "dirname": "test1",
            "hold": [
              {
                "active": true,
                "filename": "4.html",
                "id": 4,
                "name": "D - 4",
              },
              {
                "active": true,
                "filename": "5.html",
                "id": 5,
                "name": "E - 5",
              },
              {
                "active": false,
                "filename": "6.html",
                "id": 6,
                "name": "F - 6",
              },
            ],
            "name": "Test",
            "trial": [],
          }
        }
        useShortname={true}
      />
    </div>
    <div
      className="svg-radar"
    >
      <svg
        id="radar"
      />
    </div>
    <div
      className="radar-entries"
    >
      <h2>
        Test 4
      </h2>
      <QuadrantList
        blips={
          [
            {
              "active": true,
              "color": "#43a047",
              "id": 1,
              "index": 0,
              "label": "Java",
              "link": "/entries/java.html",
              "moved": false,
              "quadrant": 2,
              "ring": 0,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -14.474380723221014,
              "y": -41.11798108601353,
            },
            {
              "active": true,
              "color": "#bf360c",
              "id": 3,
              "index": 1,
              "label": "PHP",
              "link": "/entries/php.html",
              "moved": false,
              "quadrant": 2,
              "ring": 3,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -382.075380904692,
              "y": -21.17457418758074,
            },
          ]
        }
        headerLevel={3}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        quadrant={
          {
            "adopt": [
              {
                "active": true,
                "filename": "1.html",
                "id": 1,
                "name": "A - 1",
              },
              {
                "active": true,
                "filename": "2.html",
                "id": 2,
                "name": "B - 2",
              },
            ],
            "assess": [
              {
                "active": true,
                "filename": "3.html",
                "id": 3,
                "name": "C - 3",
              },
            ],
            "dirname": "test1",
            "hold": [
              {
                "active": true,
                "filename": "4.html",
                "id": 4,
                "name": "D - 4",
              },
              {
                "active": true,
                "filename": "5.html",
                "id": 5,
                "name": "E - 5",
              },
              {
                "active": false,
                "filename": "6.html",
                "id": 6,
                "name": "F - 6",
              },
            ],
            "name": "Test",
            "trial": [],
          }
        }
        useShortname={true}
      />
      <h2>
        Test 1
      </h2>
      <QuadrantList
        blips={
          [
            {
              "active": true,
              "color": "#43a047",
              "id": 1,
              "index": 0,
              "label": "Java",
              "link": "/entries/java.html",
              "moved": false,
              "quadrant": 2,
              "ring": 0,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -14.474380723221014,
              "y": -41.11798108601353,
            },
            {
              "active": true,
              "color": "#bf360c",
              "id": 3,
              "index": 1,
              "label": "PHP",
              "link": "/entries/php.html",
              "moved": false,
              "quadrant": 2,
              "ring": 3,
              "segment": {
                "clipx": [Function],
                "clipy": [Function],
                "random": [Function],
              },
              "vx": 0,
              "vy": 0,
              "x": -382.075380904692,
              "y": -21.17457418758074,
            },
          ]
        }
        headerLevel={3}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        quadrant={
          {
            "adopt": [
              {
                "active": true,
                "filename": "1.html",
                "id": 1,
                "name": "A - 1",
              },
              {
                "active": true,
                "filename": "2.html",
                "id": 2,
                "name": "B - 2",
              },
            ],
            "assess": [
              {
                "active": true,
                "filename": "3.html",
                "id": 3,
                "name": "C - 3",
              },
            ],
            "dirname": "test1",
            "hold": [
              {
                "active": true,
                "filename": "4.html",
                "id": 4,
                "name": "D - 4",
              },
              {
                "active": true,
                "filename": "5.html",
                "id": 5,
                "name": "E - 5",
              },
              {
                "active": false,
                "filename": "6.html",
                "id": 6,
                "name": "F - 6",
              },
            ],
            "name": "Test",
            "trial": [],
          }
        }
        useShortname={true}
      />
    </div>
  </div>
  <div
    className="u-cf"
  />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="one-half column"
      >
        <h2>
          About the Tech Radar
        </h2>
        <p>
          The Extenda Retail Tech Radar is based on the concept introduced by 
          <a
            href="https://www.thoughtworks.com/radar"
          >
            ThoughtWorks
          </a>
          . The Radar contains a list of technologies, complemented by an assessment result, called ring assessment. The following rings are used:
        </p>
        <ul
          className="no-bullets"
        >
          <li>
            <strong>
              Adopt
            </strong>
             â 
            Technologies we have high confidence in and recommend for wide use.
          </li>
          <li>
            <strong>
              Trial
            </strong>
             â 
            Technologies we have seen used with success, but are yet to be proved successfully in wide use and at large scale. Carries higher risk.
          </li>
          <li>
            <strong>
              Assess
            </strong>
             â 
            Technologies that are promising with a clear potential value for us. Requires research and prototyping to analyze its impact.
          </li>
          <li>
            <strong>
              Hold
            </strong>
             â 
            Technologies not recommended for new products or technology we no longer want to invest in. Existing products may still consider these depending on their legacy.
          </li>
        </ul>
      </div>
      <div
        className="one-half column"
      >
        <h2>
          The Purpose
        </h2>
        <p>
          The Tech Radar aims to inspire and support teams at Extenda Retail to pick the best technologies for their projects. It is way to share experience and knowledge between the teams and to create transparency about the technology direction of Extenda Retail. The Tech Radar also becomes a list of dos and don'ts, what to try and to avoid in order to increase success.
        </p>
        <h2>
          How to Contribute
        </h2>
        <p>
          The Tech Radar is maintained on
           
          <a
            href="https://github.com/extenda/tech-radar"
          >
            GitHub
          </a>
          . There you can read more about the process and how to contribute.
        </p>
      </div>
    </div>
  </div>
</Fragment>
`;
